Userland/Libraries/LibGUI/Window.cpp

```diff
@@ -522,6 +525,48 @@ void Window::propagate_shortcuts(KeyEvent& event, Widget* widget, ShortcutPropag
     event.ignore();
 }
 
+void Window::restore_size_and_position(StringView domain, StringView group, Optional<Gfx::IntSize> fallback_size, Optional<Gfx::IntPoint> fallback_position)
```

Comment:
I would not make the fallbacks optional, and it can just be one parameter - an `IntRect`. The code I posted in the last review was quite simpler and avoided all the magic values below.

Let's keep it simple - we can also just not make `group` default to "Window", to avoid mixing overloads and default parameters. Just one function, and make callers pass in the values they want.

Reply:
The reason I split size and position is because most of the time you want the windowserver to position your windows so the position is then undefined. Hardcoded init positions result in the newly opened windows that are not stacked.

Reply:
Fair point!

I'd still like to see the overload removed and just make FileManager pass in "Window". I don't think there's much value in avoiding that for the cost of having to mix these default params with an overload, just feels kinda awkward.

While playing around with this, I also found we have a FileManager.ini in Base/home/anon/.config/FileManager.ini with default Left/Top values. For bookkeeping, you might want to remove those 2 lines in your FileManager commit.

Reply:
Hmm, but on second thought, WindowServer would only get a say in the positioning for the first ever launch of this application, no? On subsequent launches, we'll have the last position saved?

Reply:
That is correct